Sources/NIOCore/AsyncChannel/AsyncChannel.swift

```diff
@@ -99,6 +99,61 @@ public struct NIOAsyncChannel<Inbound: Sendable, Outbound: Sendable>: Sendable {
     /// - Parameters:
     ///   - channel: The ``Channel`` to wrap.
     ///   - configuration: The ``NIOAsyncChannel``s configuration.
+    ///   - closeOnDeinit: Indicates if the underlying channel should be closed once the `inbound` and `outbound` have been deinited. We do not recommend to rely on
+    ///   deinit based resource tear down.
+    @inlinable
+    public init(
+        synchronouslyWrapping channel: Channel,
+        configuration: Configuration = .init(),
+        closeOnDeinit: Bool
```

Comment:
It's a shame we have to rely on users setting this to get the right behaviour because they can still set this to be `true` and get the old behaviour without a deprecation warning.

Ideally users shouldn't have to think about this and we would have two differently spelled `init`s, the deprecated old one and the new no-close-on-deinit one. That'd make it obvious which one to call. 


I'm not sure what to suggest for naming here, however. A couple of ideas (don't love either):
- a `static` helper `static func synchronouslyWrapping(_:configuration:) -> Self`
- `init(wrappingChannelSynchronously:configuration:)`

Reply:
I agree with those points. I had it quickly prototyped with a factory method but disliked it. Maybe just having a `wrappingChannelSynchronously` is the way out :D